clusterloader2/pkg/measurement/common/nodelocaldns_latency_prometheus.go

```diff
@@ -0,0 +1,93 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package common
+
+import (
+	"fmt"
+	"time"
+
+	"k8s.io/klog"
+	"k8s.io/perf-tests/clusterloader2/pkg/errors"
+	"k8s.io/perf-tests/clusterloader2/pkg/measurement"
+	measurementutil "k8s.io/perf-tests/clusterloader2/pkg/measurement/util"
+	"k8s.io/perf-tests/clusterloader2/pkg/util"
+)
+
+const (
+	nodelocaldnsLatencyPrometheusMeasurementName = "NodeLocalDNSLatencyPrometheus"
+	perc99LatencyQuery                           = `histogram_quantile(0.99, sum(rate(http_request_duration_seconds_bucket[%v])) by (le))`
+	defaultLatencyUpperBoundSeconds              = 5
+)
+
+func init() {
+	create := func() measurement.Measurement { return CreatePrometheusMeasurement(&nodelocaldnsLatencyGatherer{}) }
+	if err := measurement.Register(nodelocaldnsLatencyPrometheusMeasurementName, create); err != nil {
+		klog.Fatalf("Cannot register %s: %v", nodelocaldnsLatencyPrometheusMeasurementName, err)
+	}
+}
+
+type nodelocaldnsLatencyGatherer struct{}
+
+func (n *nodelocaldnsLatencyGatherer) Gather(executor QueryExecutor, startTime time.Time, config *measurement.Config) ([]measurement.Summary, error) {
+	result, err := n.get99thPercentileLatency(executor, startTime)
```

Comment:
nit: since we should gather multiple percentiles, let's call it gatherDNSLatencies or sth like that

Reply:
Called it "getPerncetileLatencies"